### PR TITLE
feat: Support SetIfNotExists

### PR DESCRIPTION
**Pull request by rlinehan**
`SetIfNotExists` stores a value for a key if the key is not already present (and returns `Stored`), otherwise it does not store the value (and returns `NotStored`).

Even though it's not really to do with "data structures," this API is only available in caches that also support data structures.

The server side changes for this have not all landed yet, so the tests won't pass in CI. I wanted to get this up for review anyway, but we should hold off merging it until this actually exists for real :grin:.

(Also, I've never written any PHP before this, so a lot of no idea what I'm doing going on here).

### CODE REVIEW

File: tests/Cache/CacheClientTest.php [hunk 1]
```diff
@@ -447,6 +447,88 @@ public function testGetTimeout()
...
+    // SetIfNotExists tests
+
+    public function testSetIfNotExists()
+    {
+        $key = uniqid();
+        $value = uniqid();
+
+        $response = $this->client->setIfNotExists($this->TEST_CACHE_NAME, $key, $value);
+        $this->assertNull($response->asError());
+        $response = $response->asStored();
```
Comment (pgautier404) [inline tests/Cache/CacheClientTest.php:459]: Do you want to an assertion about `$response` not being null here? Also, if you expect this to *not* return `NotStored` you might assert that `asNotStored` returns null.

File: src/Cache/CacheOperationTypes/CacheOperationTypes.php [hunk 2]
```diff
@@ -357,6 +368,45 @@ class CacheGetResponseError extends CacheGetResponse
...
+    {
+        if ($this->isError()) {
+            return $this;
+        }
+        return null;
+    }
+}
+
+class CacheSetIfNotExistsResponseStored extends CacheSetIfNotExistsResponse {
+}
```
Comment (pgautier404) [inline src/Cache/CacheOperationTypes/CacheOperationTypes.php:399]: Our vanilla `set` operatio returns the key and value in the response object. Is there any value in having `CacheSetIfNotExistsResponseStored` behave the same way?

Comment (rlinehan) [reply src/Cache/CacheOperationTypes/CacheOperationTypes.php:399]: Sure, I can do that for consistency.

File: tests/Cache/CacheClientTest.php (hunk 1 above)
Comment (rlinehan) [reply tests/Cache/CacheClientTest.php:459]: Whoops, thanks for catching the missing assertion!

File: src/Cache/CacheOperationTypes/CacheOperationTypes.php (hunk 2 above)
Comment (rlinehan) [reply src/Cache/CacheOperationTypes/CacheOperationTypes.php]: Done in [e454276](https://github.com/momentohq/client-sdk-php/pull/91/commits/e45427622a5cad451d229776518230f4e62e5e6f)

File: tests/Cache/CacheClientTest.php (hunk 1 above)
Comment (rlinehan) [reply tests/Cache/CacheClientTest.php:460]: Done in [374d3ff](https://github.com/momentohq/client-sdk-php/pull/91/commits/374d3ff8477a745d539875da8e48bad5c2846c88)

**Review by pgautier404 — APPROVED** [review]
LGTM!

**Review by cprice404 — APPROVED** [review]
seems like there is a CI issue but otherwise 🚢

**Comment by rlinehan** [comment]
Server side changes merged into alpha and CI is passing 🎉

**Pull request merged**